### ISSUE TITLE
[Cmake] catkin_package required for installation

### ISSUE THREAD

**Issue by paLeziart** [stack-of-tasks/tsid#65]
Dear all,

I ran into an issue trying to install TSID. It is asking for catkin-pkg but this package is not in the dependencies of [the main page.](https://github.com/stack-of-tasks/tsid#dependencies)

It seems there is some discussion going on about catkin preventing packaging in #48 involving the following lines in CMakeList.txt.
```
# Fail-safe support for catkin-ized pinocchio:
#  - If catkin-based pinocchio is installed it runs the CFG_EXTRAS to set up the Pinocchio preprocessor directives
#  - If it isn't, nothing happens and the subsequent pkg-config check takes care of everything.
find_package(catkin QUIET COMPONENTS pinocchio)
```

Looks like it is not supposed to block me if I do not have catkin_pkg.

Here is the output of my terminal:
```
cmake .. -DCMAKE_BUILD_TYPE=RELEASE -DCMAKE_INSTALL_PREFIX=/local/users/paleziart -DPYTHON_EXECUTABLE=$(which python3.5) -DPYTHON_STANDARD_LAYOUT=ON
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Performing Test R-pedantic
-- Performing Test R-pedantic - Success
-- Performing Test R-Wno-long-long
-- Performing Test R-Wno-long-long - Success
-- Performing Test R-Wall
-- Performing Test R-Wall - Success
-- Performing Test R-Wextra
-- Performing Test R-Wextra - Success
-- Performing Test R-Wcast-align
-- Performing Test R-Wcast-align - Success
-- Performing Test R-Wcast-qual
-- Performing Test R-Wcast-qual - Success
-- Performing Test R-Wformat
-- Performing Test R-Wformat - Success
-- Performing Test R-Wwrite-strings
-- Performing Test R-Wwrite-strings - Success
-- Performing Test R-Wconversion
-- Performing Test R-Wconversion - Success
-- Found Doxygen: /usr/bin/doxygen (found version "1.8.11") 
-- Option EIGEN_RUNTIME_NO_MALLOC on.
-- eigen3 >= 3.2.0 is required.
-- Checking for module 'eigen3 >= 3.2.0'
--   Found eigen3 , version 3.2.92
-- Pkg-config module eigen3 v3.2.92 has been detected with success.
-- Using CATKIN_DEVEL_PREFIX: /local/users/paleziart/tsid/_build-RELEASE/devel
-- Using CMAKE_PREFIX_PATH: /opt/ros/kinetic
-- This workspace overlays: /opt/ros/kinetic
-- Found PythonInterp: /usr/bin/python3.5 (found suitable version "3.5.2", minimum required is "2") 
-- Using PYTHON_EXECUTABLE: /usr/bin/python3.5
-- Using Debian Python package layout
-- Using empy: /usr/bin/empy
-- Using CATKIN_ENABLE_TESTING: ON
-- Call enable_testing()
-- Using CATKIN_TEST_RESULTS_DIR: /local/users/paleziart/tsid/_build-RELEASE/test_results
-- Found gtest sources under '/usr/src/gmock': gtests will be built
-- Found gmock sources under '/usr/src/gmock': gmock will be built
-- Found PythonInterp: /usr/bin/python3.5 (found version "3.5.2") 
-- Using Python nosetests: /usr/bin/nosetests
ImportError: "from catkin_pkg.package import parse_package" failed: No module named 'catkin_pkg'
Make sure that you have installed "catkin_pkg", it is up to date and on the PYTHONPATH.
CMake Error at /opt/ros/kinetic/share/catkin/cmake/safe_execute_process.cmake:11 (message):
  execute_process(/usr/bin/python3.5
  "/opt/ros/kinetic/share/catkin/cmake/parse_package_xml.py"
  "/opt/ros/kinetic/share/catkin/cmake/../package.xml"
  "/local/users/paleziart/tsid/_build-RELEASE/catkin/catkin_generated/version/package.cmake")
  returned error code 1
Call Stack (most recent call first):
  /opt/ros/kinetic/share/catkin/cmake/catkin_package_xml.cmake:74 (safe_execute_process)
  /opt/ros/kinetic/share/catkin/cmake/all.cmake:168 (_catkin_package_xml)
  /opt/ros/kinetic/share/catkin/cmake/catkinConfig.cmake:20 (include)
  CMakeLists.txt:80 (find_package)

-- Configuring incomplete, errors occurred!
See also "/local/users/paleziart/tsid/_build-RELEASE/CMakeFiles/CMakeOutput.log".
See also "/local/users/paleziart/tsid/_build-RELEASE/CMakeFiles/CMakeError.log".
```

**Comment by nim65s** [comment]
This is the issue I always have with this line, I should have been more explicit when I tried to remove it in  https://github.com/stack-of-tasks/tsid/commit/50144111d09f722e01b0fbccdc37b91b6630d9e3

I guess that parts of catkin are found, and therefore it assumes that everything is available and functionnal, while it is not. Here, we want to build with python 3, and packages from the ROS binary package repositories are not available in python 3, so catkin is found, but the python modules are not.

I think we could remove this line, and use another way to package TSID on ROS ?

**Comment by andreadelprete** [comment]
Sounds good to me!

**Comment by nim65s** [comment]
Fixed in 1.2.3.
For use with ROS (at least catkin build), #75 should do the job. If there is anyother need, please re-open :)
